### ISSUE TITLE
Make the code doc floating panel closable using the escape key

### ISSUE THREAD

**Issue by elkdanger** [Microsoft/vscode#15960]
- VSCode Version: 1.7.2
- OS Version: OSX 10.12 (Sierra)

When using Typescript, the floating code documentation window will pop up and float above the code, but I cannot close it using (naturally) the escape key. I have to use the mouse to click somewhere within the code window to then hide the popup, which is more effort and hinders typing efficiency.

It would be nice if the escape key could be used to hide this popup.

Example of the popup that I'm talking about:

![screen shot 2016-11-23 at 13 41 48](https://cloud.githubusercontent.com/assets/766403/20563936/9c8196fc-b182-11e6-9fdd-76020f09851b.png)

Steps to Reproduce:

1. When using Typescript, type some code that would access some imported class or object which has documentation comments associated with it
2. Observe that the popup cannot be closed using the escape key

**Comment by alexandrudima** [comment]
@elkdanger I cannot reproduce. When I hit Escape, and the parameter hints is shown, it goes away. Is it possible you have installed an extension that overwrites what Escape does? i.e. Can you reproduce this after disabling all extensions:

![parameter-hints](https://cloud.githubusercontent.com/assets/5047891/20749766/0160e28c-b6f4-11e6-960a-562fca5f7427.gif)

**Comment by elkdanger** [comment]
@alexandrudima You're right, the issue seems to be with [vscodevim](https://github.com/VSCodeVim/Vim) which, given the very nature of how Vim works, makes complete sense. There's a [similar issue open](https://github.com/VSCodeVim/Vim/issues/1041) dealing with intellisense, snippets and the escape key, so I'll try and interject something over there.

Cheers

**Comment by elkdanger** [comment]
Actually while I'm here, can another key be used or bound to an action which can cancel intellisense popups? Is that available in the keybindings currently?

**Comment by alexandrudima** [comment]
@elkdanger Please give our documentation a quick read :) https://code.visualstudio.com/docs/customization/keybindings#_customizing-shortcuts

TL;DR you can customize each and every keypress in each and every context. It requires a basic understanding of how the keybindings are dispatched.

I'd like to use this issue as a test to see if our documentation is good enough. Can you (given you read the explanation at that link) come up with a keybinding rule that does what you ask for? If not, I can simply write one for you, or better yet, can we try figuring out what's wrong with the documentation :).

Thank you!

**Comment by alexandrudima** [comment]
Closing until more information is provided.
